### PR TITLE
feat: Add support for customizable projectFileExtensions

### LINKED ISSUE #19527: Include `.scss` and `.sass` in `stats.json` hashes
### Describe your motivation

We use SASS (`.scss`) stylesheets under `/frontend/styles`. These are imported and fully supported by Vite. It seems that the `vaadin:stats` Vite plugin does not generate hashes for them and store in `stats.json#frontendHashes`. Because of this, even with `prod.bundle`, the Vaadin Maven/Gradle plugins think there are new files, and thus, do not use the existing bundle.

### Describe the solution you'd like

`const projectFileExtensions = ['.js', '.js.map', '.ts', '.ts.map', '.tsx', '.tsx.map', '.css', '.css.map'];` needs to also support `.scss` and `.sass`.

Is there a reason those file extensions are not already included? If not, happy to contribute this simple PR.

### Describe alternatives you've considered

I could write a Vite plugin that computes and adds the hashes.

I could create a Gradle task that compiles the SASS files and outputs them to `/frontend/styles`.

Both options seem messy.

### Additional context

I discovered this while investigating ways to speed up our builds, in lieu of Vaadin's proper Gradle caching support. https://github.com/vaadin/flow/issues/17941

This idea came from our recent endeavour to speed up builds. Relevant issues:
* https://github.com/vaadin/flow/issues/17941
* https://github.com/gradle/gradle/issues/27696
* https://github.com/vaadin/flow/issues/19527
* https://github.com/vaadin/flow/issues/19528
* https://github.com/vaadin/docs/issues/3477
* https://github.com/vaadin/flow/issues/19543

### PR DESCRIPTION
**Pull request by caalador**
Fixes #19527

### CODE REVIEW

File: flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinFlowPluginExtension.kt [hunk 1]
```diff
@@ -500,6 +511,7 @@ public class PluginEffectiveConfiguration(
             "frontendHotdeploy=${frontendHotdeploy.get()}," +
             "reactEnable=${reactEnable.get()}," +
             "cleanFrontendFiles=${cleanFrontendFiles.get()}" +
```
Comment (tltv) [inline flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinFlowPluginExtension.kt:513]: Missing comma

File: flow-server/src/main/java/com/vaadin/flow/server/InitParameters.java [hunk 2]
```diff
@@ -285,4 +285,5 @@ public class InitParameters implements Serializable {
      */
     public static final String APPLICATION_IDENTIFIER = "applicationIdentifier";
 
+    public static final String FRONTEND_EXTRA_EXTENSIONS = "frontendExtraFileExtensions";
```
Comment (tltv) [inline flow-server/src/main/java/com/vaadin/flow/server/InitParameters.java:288]: What's the correct naming pattern here, is it like this or "frontend.extra.file.extensions"? We probably should keep them more consistent. I added just new property in other PR with dot format based on majority of properties in InitParameters class.. But if camel case is more correct today, then the other PR should be changed.

**Review by tltv — CHANGES_REQUESTED** [review]
Should also gradle plugin's `PrepareFrontendInputProperties.kt` be updated for the new property?
Also `GenerateNpmBOMMojo` maven goal could need to use this new property.

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdateVite.java [hunk 3]
```diff
@@ -118,6 +122,8 @@ private void createGeneratedConfig() throws IOException {
                 .getResource(FrontendUtils.VITE_GENERATED_CONFIG);
         String template = IOUtils.toString(resource, StandardCharsets.UTF_8);
 
+        System.out.println(
+                "=== '" + options.getFrontendExtraFileExtensions() + "'");
```
Comment (mcollovati) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdateVite.java:126]:
```suggestion
        System.out.println(
                "=== '" + options.getFrontendExtraFileExtensions() + "'");
```

File: flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinFlowPluginExtension.kt [hunk 4]
```diff
@@ -282,6 +282,13 @@ public abstract class VaadinFlowPluginExtension @Inject constructor(private val
 
     public abstract val applicationIdentifier: Property<String>
 
+    /**
+     * The list of extra file extensions that are considered project files.
+     * Hashes are calculated for these files as part of detecting if a new prod
+     * bundle should be generated.
```
Comment (mcollovati) [inline flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinFlowPluginExtension.kt:288]: Is it only for prod bundle or also for dev bundle?

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdateVite.java [hunk 5]
```diff
@@ -127,14 +133,29 @@ private void createGeneratedConfig() throws IOException {
...
     }
 
+    private String getFrontendExtraFileExtensions() {
+        Optional<List<String>> frontendExtraFileExtensions = Optional
+                .ofNullable(options.getFrontendExtraFileExtensions());
+        if (frontendExtraFileExtensions.isPresent()
+                && frontendExtraFileExtensions.get().size() > 0) {
+            return frontendExtraFileExtensions.get().stream()
+                    .map(ext -> ext.replace("'", "\\'"))
+                    .map(ext -> ext.startsWith(".") ? ext : "." + ext)
```
Comment (mcollovati) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdateVite.java:153]: Does it make sense to also trim values?
For example, I can imagine configurations like 
`<frontendExtraFileExtensions>.ico, .png</frontendExtraFileExtensions>`

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdateVite.java (hunk 3 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdateVite.java]: I forgot to push the removal =P

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdateVite.java (hunk 5 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdateVite.java:151]: added trim

File: flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinFlowPluginExtension.kt (hunk 4 above)
Comment (caalador) [reply flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinFlowPluginExtension.kt:288]: both. Changed to only say new bundle

**Comment by caalador** [comment]
Added input property for the gradle plugin.

**Review by tltv — APPROVED** [review]
LGTM. After conflicts are resolved.

**Review by tltv — APPROVED** [review]

**Pull request merged**